pymc_marketing/mmm/base.py

```diff
@@ -663,242 +645,297 @@ def compute_channel_contribution_original_scale(self) -> DataArray:
-            upper_bound, lower_bound = {}, {}
-            for channel, budget in scenario["budget"].items():
-                if channel != "total":
-                    y_fit_lower, y_fit_upper = self._estimate_budget_contribution_fit(
-                        method=method, channel=channel, budget=budget
-                    )
-                    upper_bound[channel] = y_fit_upper
-                    lower_bound[channel] = y_fit_lower
-            upper_bounds.append(upper_bound)
-            lower_bounds.append(lower_bound)
-
-        # Plot all scenarios
-        for i, (scenario, upper_bound, lower_bound) in enumerate(
-            zip(scenarios, upper_bounds, lower_bounds, strict=False)
-        ):
-            color = f"C{i}"
-            offset = i * bar_width - 0.4 + bar_width / 2
-            label = f"Scenario {i+1}" if i else "Initial"
-            self._plot_scenario(
-                axes[0], scenario["budget"], label, color, offset, bar_width
-            )
-            self._plot_scenario(
-                axes[1],
-                scenario["contribution"],
-                label,
-                color,
-                offset,
-                bar_width,
-                upper_bound,
-                lower_bound,
-                True,
-            )
-
-        axes[0].set_title("Budget Optimization")
-        axes[0].set_xlabel("Budget")
-        axes[0].set_yticks(range(num_channels))
-        axes[0].set_yticklabels(
-            [k for k in sorted(base_data["budget"].keys()) if k != "total"]
-        )
-
-        axes[1].set_title("Contribution Optimization")
-        axes[1].set_xlabel("Contribution")
-        axes[1].set_yticks(range(num_channels))
-        axes[1].set_yticklabels(
-            [k for k in sorted(base_data["contribution"].keys()) if k != "total"]
-        )
-
-        fig.suptitle("Budget and Contribution Optimization", fontsize=16, y=1.18)
-        fig.legend(loc="upper center", bbox_to_anchor=(0.5, 1.15), ncol=4)
-
-        plt.tight_layout(rect=[0, 0, 1, 0.98])
+        # Iterate through each channel to fetch and store parameters
+        for channel in channels:
+            channel_info = {}
 
-        return fig
+            # Process each group of parameters (saturation and adstock)
+            for group_name, params in param_groups.items():
+                # Build dictionary for the current group of parameters
+                param_dict = {
+                    param.replace(group_name[:-7] + "_", ""): self.fit_result[param].quantile(quantile, dim=["chain", "draw"]).to_pandas().to_dict()[channel]
+                    for param in params if param in self.fit_result
+                }
+                channel_info[group_name] = param_dict
+
+            channels_info[channel] = channel_info
+
+        return channels_info
+    
+    def _channel_map_scales(self) -> dict:
+        return dict(zip(self.channel_columns, self.channel_transformer["scaler"].scale_))
```

Comment:
I think for `zip` you need to pass the `strict` parameter

Reply:
Hmm, now it's working! 
`mmm._channel_map_scales()`

<img width="681" alt="Screenshot 2024-05-21 at 18 43 11" src="https://github.com/pymc-labs/pymc-marketing/assets/59846724/607b7147-061e-4488-9582-6f93a8aadba5">